### PR TITLE
Maker RPC and some Auxiliary fix

### PR DESCRIPTION
**Pull request by rajarshimaitra**
This PR contains a collection of the following changes.

- Streamlined the data-directory structure logic. Instead of having various locations where test/production directories are used, we now have one function `get_home_dir()` which emits the test/prod data-dir and we use that everywhere.

- Removed the redundant wallet's display address function.

- Add the `maker-cli` and `makerd` binary apps. The `makerd` starts the maker server. The `maker-cli` app connects to the server and fetches info via RPC calls.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -17,8 +17,7 @@ log = "^0.4"
 futures = "0.3"
 dirs = "3.0.1"
 tokio-socks = "0.5"
-reqwest = { version = "0.11", features = ["socks"] }
-clap = { version = "4.3.19", features = ["derive"] }
+clap = { version = "3.2.22", features = ["derive"] }
```
Comment (Shourya742) [inline Cargo.toml:20]: Any reason to downgrade to v3.*.* from v4.*.*

Comment (rajarshimaitra) [reply Cargo.toml:20]: V4 Doesn't have the custom parse function capability. I don't know why they removed it, and couldn't identify an alternative.

File: tests/test_framework/mod.rs [hunk 2]
```diff
@@ -140,29 +140,30 @@ impl TestFramework {
...
             .unwrap(),
         ));
-
-        // Create the Makers as per given configuration map.
+        let mut base_rpc_port = 3500; // Random port for RPC connection in tests. (Not used)
+                                      // Create the Makers as per given configuration map.
         let makers = makers_config_map
             .iter()
             .map(|(port, behavior)| {
+                base_rpc_port += 1;
```
Comment (Shourya742) [inline tests/test_framework/mod.rs:156]: Shouldn't we first check if the port is available for use or if it is occupied by another process?

File: src/bin/maker-cli.rs [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+        Commands::SwapUtxo => {
+            send_rpc_req(&RpcMsgReq::SwapUtxo).await?;
+        }
+    }
+
+    Ok(())
+}
+
+async fn send_rpc_req(req: &RpcMsgReq) -> Result<(), MakerError> {
+    let mut stream = TcpStream::connect("127.0.0.1:8080").await?;
```
Comment (Shourya742) [inline src/bin/maker-cli.rs:87]: Shouldn't the user be allowed to connect to the makerd of their choice? We can expect an rpc_port  or a list of RPC ports, from the user and then connect to the specific TCP stream. This way, we can remove the hardcoded string

**Review by Shourya742 — CHANGES_REQUESTED** [review]
tACK. Just few nits

File: src/bin/makerd.rs [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    pub rpc: String,
+    /// Sets the rpc basic authentication.
+    #[clap(
+        name = "USER:PASSWD",
+        short = 'a',
+        long,
+        value_parser = parse_proxy_auth,
+        default_value = "user:password",
+    )]
+    pub auth: (String, String),
```
Comment (Shourya742) [inline src/bin/makerd.rs:41]: The authentication details should come from the env rather than being passed as a command line argument.

**Pull request merged**